src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -62,24 +62,35 @@ public class AppiumCommandExecutor extends HttpCommandExecutor {
 
     private AppiumCommandExecutor(Map<String, CommandInfo> additionalCommands, DriverService service,
                                   URL addressOfRemoteServer,
-                                  HttpClient.Factory httpClientFactory) {
+                                  HttpClient.Factory httpClientFactory,
+                                  ClientConfig clientConfig) {
         super(additionalCommands,
-                ClientConfig.defaultConfig()
-                        .baseUrl(Require.nonNull("Server URL", ofNullable(service)
-                                .map(DriverService::getUrl)
-                                .orElse(addressOfRemoteServer)))
-                        .readTimeout(DEFAULT_READ_TIMEOUT), httpClientFactory);
+                clientConfig != null
+                        ? clientConfig
+                        : ClientConfig.defaultConfig()
+                            .baseUrl(Require.nonNull("Server URL", ofNullable(service)
+                                    .map(DriverService::getUrl)
+                                    .orElse(addressOfRemoteServer)))
+                            .readTimeout(DEFAULT_READ_TIMEOUT),
+                httpClientFactory != null
+                        ? httpClientFactory
+                        : getDefaultClientFactory()
```

Comment:
```suggestion
                ofNullable(httpClientFactory).orElseGet(this::getDefaultClientFactory)
```

Reply:
👍 InteiJ suggested changing `this` to `HttpCommandExecutor` as static, so I followed it in bd780ce